primaryHandler.go

```diff
@@ -213,3 +224,199 @@ func authenticationHandler(v *viper.Viper, logger log.Logger, registry xmetrics.
+		}
+		defer stopWatch()
+
+		builtValidators, err := ancla.BuildValidators(webhookConfig.Validation)
+		if err != nil {
+			return fmt.Errorf("failed to initialize webhook validators: %s", err)
+		}
+
+		addWebhookHandler := ancla.NewAddWebhookHandler(svc, ancla.HandlerConfig{
+			MetricsProvider:   in.metricsRegistry,
+			V:                 builtValidators,
+			DisablePartnerIDs: webhookConfig.DisablePartnerIDs,
+			GetLogger:         getLogger,
+		})
+
+		getAllWebhooksHandler := ancla.NewGetAllWebhooksHandler(svc, ancla.HandlerConfig{
+			GetLogger: getLogger,
+		})
+
+		in.APIRouter.Handle("/hook", in.authenticate.Then(addWebhookHandler)).Methods(http.MethodPost)
+		in.APIRouter.Handle("/hooks", in.authenticate.Then(getAllWebhooksHandler)).Methods(http.MethodGet)
+		level.Info(in.logger).Log("Webhook service enabled")
+	} else {
+		level.Info(in.logger).Log(logging.MessageKey(), "Webhook service disabled")
+	}
+	return nil
+}
+
+func provideHandlers() fx.Option {
+	return fx.Provide(
+		webhookHandler,
+		authenticationProvider,
+	)
+}
+
+func statServiceProvider(in ServiceConfigIn) *stat.ServiceOptions {
+	//
+	// Stat Service configs
+	//
+	return &stat.ServiceOptions{
+		HTTPTransactor: transaction.New(
+			&transaction.Options{
+				Do: xhttp.RetryTransactor( //nolint:bodyclose
+					xhttp.RetryOptions{
+						Logger:   in.logger,
+						Retries:  in.v.GetInt(reqMaxRetriesKey),
+						Interval: in.v.GetDuration(reqRetryIntervalKey),
+					},
+					in.xmidtHTTPClient.Do),
+				RequestTimeout: in.xmidtClientTimeout.RequestTimeout,
+			}),
+		XmidtStatURL: fmt.Sprintf("%s/device/${device}/stat", in.v.GetString(targetURLKey)),
+	}
+}
+
+type ServiceConfigIn struct {
+	fx.In
+	v                  viper.Viper
+	logger             log.Logger
+	xmidtHTTPClient    *http.Client
+	xmidtClientTimeout httpClientTimeout
+}
+
+func translationOptionsProvider(in ServiceConfigIn) *translation.ServiceOptions {
+	//
+	// WRP Service configs
+	//
+	return &translation.ServiceOptions{
+		XmidtWrpURL: fmt.Sprintf("%s/device", in.v.GetString(targetURLKey)),
+		WRPSource:   in.v.GetString(wrpSourceKey),
+		T: transaction.New(
+			&transaction.Options{
+				RequestTimeout: in.xmidtClientTimeout.RequestTimeout,
+				Do: xhttp.RetryTransactor( //nolint:bodyclose
+					xhttp.RetryOptions{
+						Logger:   in.logger,
+						Retries:  in.v.GetInt(reqMaxRetriesKey),
+						Interval: in.v.GetDuration(reqRetryIntervalKey),
+					},
+					in.xmidtHTTPClient.Do),
+			}),
+	}
+}
+
+func provideServers() fx.Option {
+	return fx.Options(
+		arrangehttp.Server{
+			Name: "server_primary",
+			Key:  "primary",
+		}.Provide(),
+		arrangehttp.Server{
+			Name: "server_health",
+			Key:  "health",
+		}.Provide(),
+		arrangehttp.Server{
+			Name: "server_pprof",
+			Key:  "pprof",
+		}.Provide(),
+		arrangehttp.Server{
+			Name: "server_metrics",
+			Key:  "metric",
+		}.Provide(),
+		fx.Invoke(
+			handlePrimaryEndpoint,
```

Comment:
In the future, we should separate the part of `handlePrimaryEndpoint()` where the API router is set up from the rest of the function; this will allow us to set up the router in a `Provide()` function and easily annotate it; right now, we have many `mux.Router`s around and just need to know that the one unnamed one is this API router.